services/motion/builtin/move_request.go

```diff
@@ -518,8 +520,34 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		return nil, err
 	}
 
+	// convert obstacles of type []GeoGeometry into []Geometry
 	geomsRaw := spatialmath.GeoGeometriesToGeometries(obstacles, origin)
 
+	// convert bounding regions which are GeoGeometries into Geometries
+	boundingRegions := spatialmath.GeoGeometriesToGeometries(req.BoundingRegions, origin)
+
+	// get robot geometries
+	robotGeoms, err := b.Geometries(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(boundingRegions) > 0 {
+		// check that the robot geometries are within the bounding region
+		collisionCheck := motionplan.NewBoundingRegionConstraint(robotGeoms, boundingRegions, defaultCollisionBufferMM)
+		if !collisionCheck(&ik.State{}) {
```

Comment:
Shouldnt this take the robot state instead of using an empty `ik.State`? This is a little confusing for me

Reply:
> motionplan.NewBoundingRegionConstraint(robotGeoms, boundingRegions, defaultCollisionBufferMM)

this initializes the constraint with the "robot state" aka `robotGeoms`.
By passing in an empty `ik.State` we hit the default case of our switch statement an return the robot state as it is without any modifications.